scylla/src/transport/session.rs

```diff
@@ -899,10 +903,18 @@ impl Session {
             is_confirmed_lwt: prepared.is_confirmed_lwt(),
         };
 
-        let span = trace_span!(
-            "Request",
-            prepared_id = %format_args!("{:X}", prepared.get_id())
-        );
+        let span = RequestSpan::new_prepared(token_and_key.as_ref(), serialized_values.size());
+
+        if !span.span().is_disabled() {
```

Comment:
1. It's to record the list of replicas. This information cannot be computed for an unprepared query.
2. AFAIK, whether a span is enabled or not depends on how the user configures the tracing library. For example, you can disable all traces below given verbosity level, or provide a custom filter that analyzes the trace field names etc. This is just a small optimization.

Reply:
Again about 1.: you do the check for prepared statement only, because only there you record replicas. And only recording replicas is considered by you to be worth such an optimisation. Correct?

Reply:
Yes.